mutex.go

```diff
@@ -78,6 +78,7 @@ type mutex struct {
 	bucket  string
 	object  string
 	client  *http.Client
+	context context.Context
```

Comment:
it is discouraged to keep a context as a struct field and most of the time would be a biv exception.

context changes from one request to another, from one call to another. by doing this all users are forced to use the same context (which kind of defeats the purpose) or need to create a new lock for every new call (not ideal).